src/Search/SearchHeader.tsx

```diff
@@ -1,62 +1,64 @@
 import {StackNavigationOptions} from '@react-navigation/stack'
 import React from 'react'
-import {StyleSheet, TouchableOpacity, TouchableOpacityProps} from 'react-native'
+import {TextInput, TouchableOpacity, TouchableOpacityProps} from 'react-native'
 
 import {Icon} from '../components'
-import {SearchBar} from './SearchBar'
+import {defaultStackNavigationOptionsV2} from '../navigation'
 import {useSearch} from './SearchContext'
 
-export const useSearchHeaderOptions = ({placeHolderText, title}) => {
+export const useSearchHeaderOptions = ({
+  placeHolderText,
+  title,
+  noBack = false,
+}: {
+  placeHolderText: string
+  title: string
+  noBack?: boolean
+}) => {
   const [searchVisible, setSearchVisible] = React.useState(false)
-  const {clearSearch} = useSearch()
+  const {search, clearSearch} = useSearch()
   const handleSearchClose = () => {
     setSearchVisible(false)
     clearSearch()
   }
 
   const searchHeaderOptions: StackNavigationOptions = searchVisible
     ? {
-        headerTitleContainerStyle: styles.headerTitleContainer,
-        headerTitle: () => <SearchHeader placeholder={placeHolderText} onClose={handleSearchClose} />,
-        headerLeftContainerStyle: styles.disableFlex,
-        headerLeft: () => null,
-        headerRightContainerStyle: styles.disableFlex,
-        headerRight: () => null,
+        ...defaultStackNavigationOptionsV2,
+        headerTitle: () => <SearchHeader placeholder={placeHolderText} />,
+        headerRight: () => (search.length > 0 ? <EraseButton onPress={clearSearch} /> : null),
+        headerLeft: () => <BackButton onPress={handleSearchClose} />,
+        headerTitleAlign: 'left',
+        headerTitleContainerStyle: {
+          flex: 1,
+        },
+        headerBackTitleVisible: false,
       }
     : {
-        headerTitleContainerStyle: styles.headerTitleContainer,
+        ...defaultStackNavigationOptionsV2,
         title: title,
         headerRight: () => <SearchButton onPress={() => setSearchVisible(true)} />,
-        headerRightContainerStyle: {
-          ...styles.disableFlex,
-          paddingHorizontal: 16,
-          position: 'absolute',
```

Comment:
I remember that this was added to avoid the search to overflow top (around safeareaview)